streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -203,8 +256,17 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
   private class ReceiverTrackerActor extends Actor {
     override def receive: PartialFunction[Any, Unit] = {
       case RegisterReceiver(streamId, typ, host, receiverActor) =>
-        registerReceiver(streamId, typ, host, receiverActor, sender)
-        sender ! true
+        // Actor stops to accept registering when tracker is stopping
+        if (!isTrackerStopping) {
+          registerReceiver(streamId, typ, host, receiverActor, sender)
+          sender ! true
+        }
```

Comment:
I propose that returning false if `isTrackerStopping  == true`. And if `onReceiverRegister` receives `false`, it just throws an exception.